code/android-core-library/src/main/java/com/adobe/marketing/mobile/configuration/ConfigurationExtension.kt

```diff
@@ -0,0 +1,454 @@
+        LaunchRulesEvaluator("Configuration", LaunchRulesEngine(extensionApi), extensionApi),
+        Executors.newSingleThreadScheduledExecutor()
+    )
+
+    /**
+     * Exists only for cascading components for dependency injection.
+     */
+    private constructor(
+        extensionApi: ExtensionApi,
+        serviceProvider: ServiceProvider,
+        appIdManager: AppIdManager,
+        cacheFileService: CacheFileService,
+        launchRulesEvaluator: LaunchRulesEvaluator,
+        retryWorker: ScheduledExecutorService
+    ) : this(
+        extensionApi,
+        serviceProvider,
+        appIdManager,
+        cacheFileService,
+        launchRulesEvaluator,
+        retryWorker,
+        ConfigurationStateManager(
+            appIdManager,
+            cacheFileService,
+            serviceProvider.networkService,
+            serviceProvider.deviceInfoService,
+            serviceProvider.dataStoreService
+        ),
+        ConfigurationRulesManager(
+            launchRulesEvaluator,
+            serviceProvider.dataStoreService,
+            serviceProvider.networkService,
+            cacheFileService
+        )
+    )
+
+    @VisibleForTesting
+    internal constructor(
+        extensionApi: ExtensionApi,
+        serviceProvider: ServiceProvider,
+        appIdManager: AppIdManager,
+        cacheFileService: CacheFileService,
+        launchRulesEvaluator: LaunchRulesEvaluator,
+        retryWorker: ScheduledExecutorService,
+        configurationStateManager: ConfigurationStateManager,
+        configurationRulesManager: ConfigurationRulesManager
+    ) : super(extensionApi) {
+        this.serviceProvider = serviceProvider
+        this.appIdManager = appIdManager
+        this.cacheFileService = cacheFileService
+        this.launchRulesEvaluator = launchRulesEvaluator
+        this.retryWorker = retryWorker
+        this.configurationStateManager = configurationStateManager
+        this.configurationRulesManager = configurationRulesManager
+    }
+
+    /**
+     * Sets up the Configuration extension for the SDK by loading the appId, publishing initial state
+     * and registering relevant event listener for processing events.
+     */
+    override fun onRegistered() {
+        super.onRegistered()
+
+        // TODO: Register pre-processor
+        api.registerEventListener(
+            EVENT_TYPE,
+            EVENT_SOURCE_REQUEST_CONTENT,
+            extensionEventListener
+        )
+
+        val appId = this.appIdManager.loadAppId()
+
+        if (!appId.isNullOrBlank()) {
+            val eventData: MutableMap<String, Any?> =
+                mutableMapOf(
+                    CONFIGURATION_REQUEST_CONTENT_JSON_APP_ID to appId,
+                    CONFIGURATION_REQUEST_CONTENT_IS_INTERNAL_EVENT to true
+                )
+            dispatchConfigurationRequest(eventData)
+        }
+
+        val initialConfig: Map<String, Any?> = this.configurationStateManager.loadInitialConfig()
```

Comment:
Can we add a verbose log here on whether the bundled/cached config is loaded or not?

Reply:
Added logs inside ConfigurationStateManager to show what config is being loaded.